### PR TITLE
Refactor Javascript Code

### PR DESCRIPTION
**Pull request by spiderxm**

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -39,4 +39,4 @@
 - [ ] This Pull Request relates to only one issue or only one feature
 - [ ] I have referenced the corresponding issue number in my commit message
 - [ ] I have added the relevant documentation.
-- [ ] My branch is up-to-date with the Upstream master branch.
+- [ ] My branch is up-to-date with the Upstream master branch.
```
Comment (dhirensr) [inline PULL_REQUEST_TEMPLATE.md:42]: what's the change here?

File: web/static/js/explore.js [hunk 2]
```diff
@@ -2,255 +2,254 @@
...
-    }
-  });
+    $.ajax({
+        type: "GET",
+        url: "/api/core/list/modules",
+        data: {},
+        success: function (result, status, xhr) {
+            let tableHtml = '<option value=\"\"> All Modules </option>';
+            for (let i = 0; i < result.length; i++) {
+                const module_name = result[i];
```
Comment (dhirensr) [inline web/static/js/explore.js:12]: should be let write as every iteration it's changing the value?

File: web/static/js/explore.js [hunk 3]
```diff
@@ -2,255 +2,254 @@
...
+                tableHtml += "<option value=" +
+                    module_name + ">"
+                    + module_name
+                    + "</option>";
+            }
+            $('#module_names').html(tableHtml);
+        },
+        error: function (jqXHR, textStatus, errorThrown) {
+            document.getElementById('error_msg').innerHTML = jqXHR.responseText;
+            if (errorThrown === "BAD REQUEST") {
```
Comment (dhirensr) [inline web/static/js/explore.js:22]: shouldn't this be removed completely as it's not doing anything?

File: web/static/js/explore.js [hunk 4]
```diff
@@ -2,255 +2,254 @@
...
+            if ($("#datatable_filter > label > input")[0].value) {
+                conf.data.filter = $("#datatable_filter > label > input")[0].value;
+            }
+
+            // Provide the same `data` options as DataTables.
+            if (typeof conf.data === 'function') {
+                // As a function it is executed with the data object as an arg
+                // for manipulation. If an object is returned, it is used as the
+                // data object to submit
+                const d = conf.data(request);
```
Comment (dhirensr) [inline web/static/js/explore.js:168]: shouldn't the variable name be descriptive other than "d"? something like data or manipulation_data

File: web/static/js/main.js [hunk 5]
```diff
@@ -299,23 +283,23 @@ function plot_event_count_by_date(event_type) {
             },
             error: function (jqXHR, textStatus, errorThrown) {
                 document.getElementById('error_msg').innerHTML = jqXHR.responseText;
-                if (errorThrown == "BAD REQUEST") {
+                if (errorThrown === "BAD REQUEST") {
```
Comment (dhirensr) [inline web/static/js/main.js:286]: same for this, can be removed if nothing is done after catching error

File: web/static/js/main.js [hunk 6]
```diff
@@ -125,13 +116,6 @@ function get_top_ten_element_in_event(event_type, element, html_element_id) {
             window.myPolarArea = Chart.PolarArea(ctx, top_ten_graph_config);
             const downloadButton = ctx.nextSibling.nextSibling;
             downloadButton.hidden = false;
-},
-        error: function (jqXHR, textStatus, errorThrown) {
```
Comment (dhirensr) [inline web/static/js/main.js:129]: error function needs to be there,maybe log or something to understand what went wrong

Comment (spiderxm) [reply web/static/js/main.js:129]: ok will add it

**Review by dhirensr — APPROVED** [review]
@Ali-Razmjoo : please review and push!

**Pull request merged**